internal/service/withdraw_extra_info.go

```diff
@@ -0,0 +1,79 @@
+package service
+
+import (
+	"context"
+	"encoding/hex"
+
+	"github.com/nervina-labs/cota-nft-entries-syncer/internal/biz"
+	"github.com/nervina-labs/cota-nft-entries-syncer/internal/data"
+	"github.com/nervina-labs/cota-nft-entries-syncer/internal/logger"
+	ckbTypes "github.com/nervosnetwork/ckb-sdk-go/types"
+)
+
+var _ Service = (*WithdrawExtraInfoService)(nil)
+
+type WithdrawExtraInfoService struct {
+	extraInfoUsecase *biz.WithdrawExtraInfoUsecase
+	logger           *logger.Logger
+	client           *data.CkbNodeClient
+}
+
+func NewWithdrawExtraInfoService(extraInfoUsecase *biz.WithdrawExtraInfoUsecase, logger *logger.Logger, client *data.CkbNodeClient) *WithdrawExtraInfoService {
+	return &WithdrawExtraInfoService{
+		extraInfoUsecase: extraInfoUsecase,
+		logger:           logger,
+		client:           client,
+	}
+}
+
+func (s WithdrawExtraInfoService) Start(ctx context.Context, _ string) error {
+	s.logger.Info(ctx, "withdraw extra info service started")
+	queryInfos, err := s.extraInfoUsecase.FindAllQueryInfos(ctx)
+	if err != nil {
+		return err
+	}
+	var block *ckbTypes.Block
+	var lock biz.Script
+	for _, v := range queryInfos {
+		block, err = s.client.Rpc.GetBlockByNumber(ctx, v.BlockNumber)
+		if err != nil {
+			return err
+		}
+		for _, tx := range block.Transactions {
+			for _, input := range tx.Inputs {
+				if hex.EncodeToString(input.PreviousOutput.TxHash[12:]) == v.OutPoint[:40] {
```

Comment:
`hex.EncodeToString(input.PreviousOutput.TxHash[12:]) == v.OutPoint[:40]`  It is recommended to extract it into a method.